pkg/internalapis/featurestates/featurestates.go

```diff
@@ -376,8 +376,15 @@ func namespaceUpdated(oldObj, newObj interface{}) {
 	_, labelPresentInNewNamespace := newNamespace.Labels[WorkLoadNamespaceLabelKey]
 
 	if !labelPresentInOldNamespace && labelPresentInNewNamespace {
+		// Label with Key vSphereClusterID is added on the namespace
 		log.Infof("Observed new workload namespace: %v", newNamespace.Name)
 		pendingCRUpdatesObj.enqueueFeatureStateUpdatesForWorkloadNamespace(ctx, newNamespace.Name)
+	} else if labelPresentInOldNamespace && !labelPresentInNewNamespace {
```

Comment:
There is no use case for this as all the namespace CRUD operations are done via WCP service APIs. But its fine if you want to handle it as I do not see it causing any harm.